src/Build.UnitTests/BinlogVersionCompatibilityTests.cs

```diff
@@ -0,0 +1,372 @@
+                throw;
+            }
+        }
+
+        private bool BuildProjectWithBinlog(string binlogParameters)
+        {
+            try
+            {
+                var logger = new BinaryLogger();
+                logger.Parameters = binlogParameters;
+
+                using var buildManager = new Microsoft.Build.Execution.BuildManager();
+                var buildParameters = new Microsoft.Build.Execution.BuildParameters
+                {
+                    Loggers = new ILogger[] { logger }
+                };
+
+                var buildRequestData = new Microsoft.Build.Execution.BuildRequestData(
+                    _testProjectPath,
+                    new Dictionary<string, string>(),
+                    null,
+                    new[] { "Build" },
+                    null);
+
+                var result = buildManager.Build(buildParameters, buildRequestData);
+                return result.OverallResult == Microsoft.Build.Execution.BuildResultCode.Success;
+            }
+            catch (Exception ex)
+            {
+                _output.WriteLine($"Build failed: {ex.Message}");
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Test that verifies binlog files have expected structure and content.
+        /// </summary>
+        [Fact]
+        public void VerifyBinlogStructure()
+        {
+            // Generate a single binlog for structure verification
+            string binlogPath = Path.Combine(_tempDirectory, "structure_test.binlog");
+            bool buildSuccess = BuildProjectWithBinlog($"LogFile={binlogPath};ProjectImports=Embed");
+            
+            buildSuccess.ShouldBeTrue("Build should succeed");
+            File.Exists(binlogPath).ShouldBeTrue("Binlog should exist");
+
+            // Test basic file format
+            var fileInfo = new FileInfo(binlogPath);
+            fileInfo.Length.ShouldBeGreaterThan(100, "Binlog should have substantial content");
+
+            // Read the first few bytes to verify it's a valid binlog format
+            using var stream = File.OpenRead(binlogPath);
+            var buffer = new byte[20];
+            int bytesRead = stream.Read(buffer, 0, buffer.Length);
+            
+            bytesRead.ShouldBeGreaterThan(0, "Should be able to read from binlog");
+            
+            // Verify the file can be opened with BinaryLogReplayEventSource
+            var reader = new BinaryLogReplayEventSource();
+            var eventCount = 0;
+            reader.AnyEventRaised += (s, e) => eventCount++;
+            reader.Replay(binlogPath); // Should not throw
+            eventCount.ShouldBeGreaterThan(0, "Should have received events from binlog");
+        }
+
+        /// <summary>
+        /// Performance test to ensure binlog generation and playback performs reasonably.
+        /// </summary>
+        [Fact]
+        public void BinlogPerformanceTest()
```

Comment:
this is not the point of the compatibility tests. Perf tests do not belong here.